### ISSUE TITLE
Unexpected behavior when isEnabled in Custom Strategy definition has an exception thrown

### ISSUE THREAD

**Issue by sheastyer** [Unleash/unleash-proxy#79]
### Describe the bug

If you create a custom strategy, and try to access a property of `undefined` during the `isEnabled` method, the entire call for getting features given a context returns a 404 - which is a very strange response. I would expect it to at least log the error, or throw a 500. Had me chasing reverse proxy/cache-ing issues for three-four hours 😂

### Steps to reproduce the bug

Create a custom strategy that accesses a property of undefined. Not the `notAField` in the return statement, this will surely throw a 'cannot access property `accountId` of undefined

You could also probably just throw an Error in the function as well and get the same result, though I did not test that. 

```
const { Strategy } = require('unleash-client');

class AccountsStrategy extends Strategy {
    constructor() {
        super('Accounts');
    }

    isEnabled(parameters, context) {
            return parameters.accountIds.includes(context.notAField.accountId);
    }
}

module.exports.AccountsStrategy = AccountsStrategy;
```

Now, register that custom strategy, and make a call to the proxy with a feature configured to be using that custom strategy. 

The app will respond with a 404 Not Found error. 

### Expected behavior

I'd expect this to fail a little more noticeably, or at least log the exception and return `false` for the feature. Not a random 404 that has nothing to do with the logic of the code. 

### Logs, error output, etc.

```bash
Setting log level to debug would show the context of the user's request before returning a 404, so I knew it was getting routed properly. 


INFO: Get enabled feature toggles for provided context [
  {
    appName: 'myApp',
    environment: 'production',
    userId: 'myUserId',
    remoteAddress: '::1',
    properties: { accountId: 'validAccountId' }
  }
]
```
```


### Screenshots

_No response_

### Additional context

This happened on the Node.js version of the self-hosted proxy, not the Docker one. Was able to reproduce in the cloud as well as locally. 

### Unleash version

latest

### Subscription type

Open source

### Hosting type

Self-hosted

### SDK information (language and version)

Was connecting to the proxy via the Javascript client SDK

**Comment by sighphyre** [comment]
Hi sheastyer, thanks for raising this, I'll have a chat with the team about this one and get back to you.

**Comment by thomasheartman** [comment]
Hey, @sheastyer! Thanks for opening the ticket and great catch! I think this will be solved by #80 which has something similar going on. The reason is that the error handler only targets a specific kind of error, leading to unknown errors turning into random 404s. The changes in #80 should be enough to catch the errors that you're seeing here as well 🤞🏼

**Comment by thomasheartman** [comment]
@sheastyer We've just released v0.10.2 now. I can confirm that the changes in #80 do address the issue you reported. As of today, a crash in a strategy will return a 500 error (the following image shows the result in the OpenAPI spec for now).

I'll close this  issue for now in the belief that it _is_ resolved. If you think it still occurs, please feel free to reopen the issue ☺️ 

![API response showing a 500 with the JSON response '"error": "Whoops! We dropped the ball on this one (an unexpected error occurred): Cannot read properties of undefined (reading 'includes')"' ](https://user-images.githubusercontent.com/17786332/179202181-43771bf9-1fa8-41d0-8744-7669efda35a0.png)

**Comment by sheastyer** [comment]
Thank you much @thomasheartman !
